src/registries/npm-registry.js

```diff
@@ -222,13 +222,9 @@ export default class NpmRegistry extends Registry {
   }
 
   getScope(packageIdent: string): string {
-    // Matches '@scope' in a packageName or pathname, otherwise returns ''
-    if (packageIdent && isScopedPackage(packageIdent)) {
-      const matched = packageIdent.match(/(^@|(?!\/)@).+?((?=%2f)|(?=\/))/);
-      return (matched && matched[0]) || '';
-    }
-
-    return '';
+    // Matches strings beginning with @ or contain /@ up until the next forward slash
+    const match = packageIdent.replace('%2f', '/').match(/(^@|\/@)(.+?)(?=\/)/);
```

Comment:
* It's only the forward slashes in the package name (eg (`https://private.npm.repo.com/@foo%2fbar`) and It would be good to find where the pathname is getting stored/retrieved from to see if it is necessary to keep it in that format. I assume it is related to [this](https://github.com/yarnpkg/yarn/blob/master/src/registries/npm-registry.js#L72).
* I'm using the `?` after a quantifier for [non-greedy matching](https://developer.mozilla.org/en/docs/Web/JavaScript/Guide/Regular_Expressions#special-questionmark), so it matches only up to the next forward slash that is in a lookahead so it doesn't get captured.
* I haven't seen a query string in a pathname yet, so it is not something I am catering for. Is that possible? Do you have some examples I could add to the test cases?

Your regex passes the test cases so I'll update it with that, with the added benefit of you having considered the query strings. :+1: